server/routes.js

```diff
@@ -0,0 +1,254 @@
+var _fs             = require("fs")
+let denodeify       = require("denodeify")
+let config          = require("../config")
+let users           = require("../users")
+
+const fs = {
+	readFile: denodeify(_fs.readFile)
+}
+
+module.exports = [
+	{
+		path:/^\/$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: ["/"],
+		context: (_, db) =>
+			aggregatePosts(db, { $sort: { timestamp: -1 } }, { $limit: 5 })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return db.posts.count()
+						.then((count) => ({
+							title: "",
+							posts,
+							pagination: {
+								right: count > 5 ? "/archive/2" : undefined
+							}
+						}))
+				})
+				.then(fillDefaultSidebar(db))
+	},
+	{
+		path:/^\/archive\/([1-9][0-9]*)$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: range(5).map((i) => `/archive/${i + 1}`),
```

Comment:
The original renderer also only prerendered the first five pages, which is why that's the default here.  (And, notably, it actually wouldn't let you access anything beyond there; that is not the case with the new renderer.)

I also just stuck with 5 because I deemed it unlikely that anyone would go further back than that in most cases, and if they do, a 50ms delay for the first person to do so (since the result will get cached) seemed reasonable.

That said, there's no reason that we couldn't have this scale based on available post count.